mindmeld/active_learning/data_loading.py

```diff
@@ -300,13 +323,26 @@ def sample_and_update(
         if confidence_segments:
             params_rank_3d["confidence_segments"] = confidence_segments
 
-        ranked_indices = (
+        ranked_indices_2d = (
             heuristic.rank_3d(**params_rank_3d)
             if confidences_3d
             else heuristic.rank_2d(confidences_2d)
         )
-        newly_sampled_indices = ranked_indices[:sampling_size]
-        remaining_indices = ranked_indices[sampling_size:]
+
+        newly_sampled_indices = ranked_indices_2d[:sampling_size]
+        remaining_indices = ranked_indices_2d[sampling_size:]
+
+        if entity_tuning:
+            ranked_entity_indices = heuristic.rank_entities(entity_confidences)
+            newly_sampled_indices_entity = ranked_entity_indices[:sampling_size]
+            remaining_indices_entity = ranked_entity_indices[sampling_size:]
+
+            newly_sampled_indices = list(
```

Comment:
By taking the union here we would get a variable batch_size between 100 (all selections overlap) to 200 (no selections overlap). In the config, we have the defined param of "batch_size" and the user would expect this value to be used. Moreover, a consistent batch_size has been important for post-processing scripts. The solution here can be to take a random sample of the union data such that only batch_size num queries are selected.